src/components/Settings/index.tsx

```diff
@@ -41,21 +41,21 @@ const MenuFlyout = styled(AutoColumn)`
   padding: 1rem;
 `
 
-export default function SettingsTab({ autoSlippage }: { autoSlippage: Percent }) {
-  const { chainId } = useWeb3React()
-  const showDeadlineSettings = Boolean(chainId && !L2_CHAIN_IDS.includes(chainId))
+export default function SettingsTab({ autoSlippage, chainId }: { autoSlippage: Percent; chainId?: number }) {
+  const { chainId: connectedChainId } = useWeb3React()
+  const showDeadlineSettings = Boolean(connectedChainId && !L2_CHAIN_IDS.includes(connectedChainId))
```

Comment:
i think this shoudl still technically be `chainId` since this is related to the component's chain and not the wallet chain lulz